utils/responsewriter/response_writer.go

```diff
@@ -154,26 +144,3 @@ func (rw *responseWriter) callBefore() {
 		rw.beforeFuncs[i](rw)
 	}
 }
-
-func (rw *responseWriter) SetUserValue(key any, value any) {
```

Comment:
Using a property in the context allows us to get rid of some maps and improve allocations and perf.